extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java

```diff
@@ -205,14 +223,28 @@ public void handle(GrpcClientResponse<Buffer, Buffer> response) {
                         response.exceptionHandler(new Handler<>() {
                             @Override
                             public void handle(Throwable t) {
-                                exporterMetrics.addFailed(numItems);
-                                logger.log(
-                                        Level.SEVERE,
-                                        "Failed to export "
-                                                + type
-                                                + "s. The stream failed. Full error message: "
-                                                + t.getMessage());
-                                result.fail();
+                                if (attemptNumber <= MAX_ATTEMPTS) {
+                                    // retry
+                                    initiateSend(client, server,
+                                            MAX_ATTEMPTS - attemptNumber,
```

Comment:
Won't we do more than 3 retries in the end, because we are calling multiple times the initiateSend() with different retry values on the same flow?

Reply:
Nope